platforms/nuttx/src/px4/stm/stm32_common/io_pins/io_timer.c

```diff
@@ -86,8 +86,11 @@ static int io_timer_handler7(int irq, void *context, void *arg);
  * We also allow for overrides here but all timer register usage need to be
  * taken into account
  */
-#if !defined(BOARD_PWM_FREQ)
+#if !defined(BOARD_PWM_FREQ) && !defined(CONFIG_MODULES_SC_RATE_CONTROL)
```

Comment:
At a minimum let's do this via kconfig and not introduce module specific ifdefs.